aws/resource_aws_ecrpublic_repository.go

```diff
@@ -0,0 +1,357 @@
+package aws
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecrpublic"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsEcrPublicRepository() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEcrPublicRepositoryCreate,
+		Read:   resourceAwsEcrPublicRepositoryRead,
+		Update: resourceAwsEcrPublicRepositoryUpdate,
+		Delete: resourceAwsEcrPublicRepositoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"repository_name": {
```

Comment:
I noticed the private ecr repository resource uses `name` rather than `repository_name`. Should be we be consistent over those resources, or consistent with the API?